lib/iex/lib/iex/introspection.ex

```diff
@@ -39,112 +30,183 @@ defmodule IEx.Introspection do
   @doc """
   Prints the documentation for the given function
   with any arity in the list of modules.
+
+  Used to find functions that either iex commands or kernel functions.
   """
   def h(modules, function) when is_list(modules) and is_atom(function) do
-    result =
-      Enum.find_value modules, false, fn module ->
-        h_mod_fun(module, function) == :ok
-      end
-
-    unless result, do: nodocs(function)
+    helpers = IEx.Config.doc_helpers(:helpers)
+    result = Enum.find_value(modules,
+               fn(module) ->
+                  Enum.find_value(helpers,
+                        fn(mod) -> found_help(mod, module, function) end)
+               end)
```

Comment:
I struggle how to format those kinds of nested iterations a lot. Are there
any examples
elsewhere in the code base ?

- Booker C. Bense

On Saturday, October 3, 2015, eksperimental <notifications@github.com>
wrote:

> In lib/iex/lib/iex/introspection.ex
> <https://github.com/elixir-lang/elixir/pull/3640#discussion_r41086777>:
>
> >    """
> >    def h(modules, function) when is_list(modules) and is_atom(function) do
> > -    result =
> > -      Enum.find_value modules, false, fn module ->
> > -        h_mod_fun(module, function) == :ok
> > -      end
> > -
> > -    unless result, do: nodocs(function)
> > +    helpers = IEx.Config.doc_helpers(:helpers)
> > +    result = Enum.find_value(modules,
> > +               fn(module) ->
> > +                  Enum.find_value(helpers,
> > +                        fn(mod) -> found_help(mod, module, function) end)
> > +               end)
>
> I find these indentation particularly hard to follow (it's just a personal
> choice), but i think the most prevalent way to identing in the Elixir
> project is just to add 2 spaces on a new level on the next line.
> having an odd number of spaces also make it diffucult to line up using the
> tab space in editors that replace a tab with two spaces.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/3640/files#r41086777>.
>